### ISSUE TITLE
train with my own dataset, what files should be prepared？

### ISSUE THREAD

**Issue by Pratyeka** [ShaoqingRen/faster_rcnn#42]
Hi all:
I would like to train faster-rcnn on my own dataset, but i do not know what data files should be prapared？? i.e. the images, annotations?   Any reference blogs or tutorials will be highly appreciated. Thanks

**Comment by rayzhang0346** [comment]
I setup a .mat with imdb and roidb for both training and testing datas according to the structure of VOC sample in imdb/cache.

**Comment by duygusar** [comment]
@rayzhang0346  I have cleaned the cache and converted my data and annotations into VOC format along with image sets list etc. But then I get this error: "proposal_prepare_image_roidb>scale_rois (line 110) Non-singleton dimensions of the two input arrays must match each other." I don't see why this is not enough, why did you set a .mat for imdb and roidb? Is it because your data format is different than VOC?

**Comment by rayzhang0346** [comment]
@duygusar Yes, i read through the code and find the key difference is VOC devkit. So i write code to deal with my data & notations and output the same format as imdb & roide. That's how I skip the data_prepare process.
I am afraid i can't help you with your error.

**Comment by duygusar** [comment]
@rayzhang0346  That's alright, thanks for your answer. Well I was hoping I wouldn't run into problems since I converted my dataset to VOC format instead. Anyhow, I will try to see what's wrong

**Comment by fengyuxi55** [comment]
@duygusar , I got same error as yours, and I don't know what's going wrong.
If you find anything, plz tell me

**Comment by duygusar** [comment]
@fengyuxi55 I have just remembered I have the same problem while training a DPM too, I basically had to rewrite the whole xml parser part because it was not able to retrieve inner tags.

**Comment by fengyuxi55** [comment]
@duygusar , thanks for your advice. I rewrite the xml parser to read my annotation files, the error disappear. Now I'm running the training code to train my own data.

**Comment by smajida** [comment]
Is anybody has trained faster-RCNN on another dataset?

**Comment by smajida** [comment]
@duygusar , I would like to retrain faster-rcnn using the the following dataset:
http://www.svcl.ucsd.edu/projects/anomaly/dataset.htm
could you tell me how to start to do so?
